src/mod_ping.erl

```diff
@@ -68,69 +72,76 @@ route_ping_iq(JID, Server) ->
-    mongoose_metrics:ensure_metric(Host, [mod_ping, ping_response_time], histogram).
+ensure_metrics(HostType) ->
+    mongoose_metrics:ensure_metric(HostType, [mod_ping, ping_response], spiral),
+    mongoose_metrics:ensure_metric(HostType, [mod_ping, ping_response_timeout], spiral),
+    mongoose_metrics:ensure_metric(HostType, [mod_ping, ping_response_time], histogram).
 
 %%====================================================================
 %% gen_mod callbacks
 %%====================================================================
 
-start(Host, Opts) ->
-    ensure_metrics(Host),
+start(HostType, Opts) ->
+    ensure_metrics(HostType),
     SendPings = gen_mod:get_opt(send_pings, Opts, ?DEFAULT_SEND_PINGS),
     IQDisc = gen_mod:get_opt(iqdisc, Opts, no_queue),
-    mod_disco:register_feature(Host, ?NS_PING),
-    gen_iq_handler:add_iq_handler(ejabberd_sm, Host, ?NS_PING,
-                                  ?MODULE, iq_ping, IQDisc),
-    gen_iq_handler:add_iq_handler(ejabberd_local, Host, ?NS_PING,
-                                  ?MODULE, iq_ping, IQDisc),
-    maybe_add_hooks_handlers(Host, SendPings).
+    %% TODO: update code related to mod_disco
+    mod_disco:register_feature(HostType, ?NS_PING),
+    gen_iq_handler:add_iq_handler_for_domain(HostType, ?NS_PING, ejabberd_sm,
+                                             fun ?MODULE:iq_ping/5, #{}, IQDisc),
+    gen_iq_handler:add_iq_handler_for_domain(HostType, ?NS_PING, ejabberd_local,
+                                             fun ?MODULE:iq_ping/5, #{}, IQDisc),
+    maybe_add_hooks_handlers(HostType, SendPings).
 
 maybe_add_hooks_handlers(Host, true) ->
     ejabberd_hooks:add(hooks(Host));
 maybe_add_hooks_handlers(_, _) ->
     ok.
 
-stop(Host) ->
+stop(HostType) ->
 %%    a word of warning: timers are installed in c2s processes, so stopping mod_ping
 %%    won't stop currently running timers. They'll run one more time, and then stop.
-    ejabberd_hooks:delete(hooks(Host)),
-    gen_iq_handler:remove_iq_handler(ejabberd_local, Host, ?NS_PING),
-    gen_iq_handler:remove_iq_handler(ejabberd_sm, Host, ?NS_PING),
-    mod_disco:unregister_feature(Host, ?NS_PING).
+    ejabberd_hooks:delete(hooks(HostType)),
+    gen_iq_handler:remove_iq_handler_for_domain(HostType, ?NS_PING, ejabberd_local),
+    gen_iq_handler:remove_iq_handler_for_domain(HostType, ?NS_PING, ejabberd_sm),
+    %% TODO: update code related to mod_disco
+    mod_disco:unregister_feature(HostType, ?NS_PING).
 
 -spec config_spec() -> mongoose_config_spec:config_section().
 config_spec() ->
     #section{
        items = #{<<"send_pings">> => #option{type = boolean},
                  <<"ping_interval">> => #option{type = integer,
-                                                validate = positive},
+                                                validate = positive,
+                                                process = fun timer:seconds/1},
```

Comment:
Why? I really want to reduce the usage of `process` as much as possible. Either unit is fine for me, just keep it the same and consistent. We have multiple timeouts configured in many options and here out of the sudden for one random option there is an unexpected conversion.

Reply:
Ok we talked it through and I think it's not an issue, let's keep the change :)